app/src/main/java/fr/free/nrw/commons/media/ZoomableActivity.kt

```diff
@@ -609,18 +598,18 @@ class ZoomableActivity : BaseActivity() {
                 .setProgressBarImage(ProgressBarDrawable())
                 .setProgressBarImageScaleType(ScalingUtils.ScaleType.FIT_CENTER)
                 .build()
-            photo!!.hierarchy = hierarchy
-            photo!!.setAllowTouchInterceptionWhileZoomed(true)
-            photo!!.setIsLongpressEnabled(false)
-            photo!!.setTapListener(DoubleTapGestureListener(photo))
+            binding.zoomable!!.hierarchy = hierarchy
+            binding.zoomable!!.setAllowTouchInterceptionWhileZoomed(true)
+            binding.zoomable!!.setIsLongpressEnabled(false)
+            binding.zoomable!!.setTapListener(DoubleTapGestureListener(binding.zoomable))
```

Comment:
This might be a good spot to make use of Kotlin's `with` syntax - avoids some of the repetition perhaps.  
```kotlin
with(binding.zoomable!!) {
    setHierarchy(hierarchy) 
    setAllowTouchInterceptionWhileZoomed(true)
    setIsLongpressEnabled(false)
    setTapListener(DoubleTapGestureListener(this))
}
```
(note: I switched to `setHierarchy()` to avoid a name collision)

What do you think?

Reply:
I think it works well, making the changes